hash_practice/exercises.py

```diff
@@ -2,18 +2,50 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(nklog(k)), where k is the length of the string 
+        Space Complexity: O(log(n))
     """
-    pass
+    if not strings:
+        return []
+
+    result = {}
+    for word in strings:
+        key = ''.join(sorted(word))
+        if key in result:
+            result.get(key).append(word)
+        else:
+            result[key] = [word]
+    return result.values()

```

Comment:
Unfortunately, this does not return a list, which is why the automated tests are failing. You can get a list from a call to `values()` by wrapping this in the `list()` function.

I'm a bit of a stickler for the automated tests passing, so I'll have to mark this Yellow for now, but just fix this bug and this part will be good enough for Green.